enum ECommand {
	LIGHT_ON = 0;
	LIGHT_OFF = 1;
	BEACON = 2;
}

message NodeAddress {
	//required fixed32 lsb = 1;
	//required fixed32 msb = 2;
	required uint32 addr = 3;
}

message SimpleCommand {
	required NodeAddress node_id = 1;
	required ECommand command = 2;
	required uint32 number = 3;
}

message SimpleAnswer {
	required NodeAddress node_id = 1;
	required ECommand command = 2;
	required uint32 number = 3;
	required uint32 answer = 4;
}

message DebugPrint {
	required string what = 1;
	//optional NodeAddress node_id = 2;
}

message PositionNotify
{
	required NodeAddress node_id = 1;
	required uint32 number = 2;
	required uint32 x = 3;
	required uint32 y = 4;
}

message MessagePackage
{
	optional PositionNotify position_notify = 1;
	optional SimpleCommand simple_command = 2;
	optional SimpleAnswer simple_answer = 3;
	optional DebugPrint debug_print = 4;
}
